src/CoreBundle/Bridge/Chameleon/Migration/Service/MainMenuMigrator.php

```diff
@@ -252,50 +246,87 @@ public function getMainCategoryMapping(): array
         return $this->mainCategoryMapping;
     }
 
+    /**
+     * Creates new main menu items for all table entries in old content boxes for which no main menu item exists yet.
+     * This method only works for standard content boxes and content boxes provided by $additionalMainCategoryMapping.
+     * It is expected that all required main menu categories already exist when this method is called.
+     *
+     * @param array $additionalMainCategoryMapping Mapping from old content box system names to new main menu category
```

Comment:
the new main menu categories must already exist?

Reply:
Yes, and it can then be given as argument.